jolt-core/src/host/mod.rs

```diff
@@ -254,6 +257,16 @@ impl Program {
     fn linker_path(&self) -> String {
         format!("/tmp/jolt-guest-linkers/{}.ld", self.guest)
     }
+
+    /// Saves the proof to a file
+    pub fn save_elf(&self) {
+        if !PathBuf::from("elf/").exists() {
+            fs::create_dir_all("elf/").expect("could not create elf directory");
+        }
+        let elf = self.elf.as_ref().unwrap();
+        fs::copy(elf, format!("elf/{}.elf", self.guest))
```

Comment:
Could you explain the rationale of this function? If we already have the ELF at the `self.elf` path, why copy it to a new directory?

Reply:
The elf file is saved to a temporary directory during proving and is then removed at the end of execution. The intention here was to have a method that could be called during execution to save the elf file to a non temporary directory. In retrospect there is not a need for a new directory to be created the project current directory is fine.

Reply:
Running the default example it does not appear to persist.

<img width="183" alt="Screenshot 2024-06-28 at 09 57 52" src="https://github.com/a16z/jolt/assets/33334338/b02c9ba5-facc-4e4f-bb28-e7e85187dbd1">

https://github.com/a16z/jolt/blob/main/jolt-core/src/host/mod.rs#L123